qctrlopencontrols/driven_controls/driven_control.py

```diff
@@ -65,80 +67,59 @@ class DrivenControl:
         self.name = name
-        if self.name is not None:
-            self.name = str(self.name)
 
-        check_none_values = [
-            (rabi_rates is None),
-            (azimuthal_angles is None),
-            (detunings is None),
-            (durations is None),
-        ]
-        all_are_none = all(value is True for value in check_none_values)
-        if all_are_none:
+        # set default values if all inputs are ``None``
+        if all(v is None for v in [rabi_rates, azimuthal_angles, detunings, durations]):
             rabi_rates = np.array([np.pi])
             azimuthal_angles = np.array([0.0])
             detunings = np.array([0.0])
             durations = np.array([1.0])
-        else:
-            # some may be None while others are not
-            input_array_lengths = []
-            if not check_none_values[0]:
-                rabi_rates = np.array(rabi_rates, dtype=np.float).reshape((-1,))
-                input_array_lengths.append(rabi_rates.shape[0])
-
-            if not check_none_values[1]:
-                azimuthal_angles = np.array(azimuthal_angles, dtype=np.float).reshape(
-                    (-1,)
-                )
-                input_array_lengths.append(len(azimuthal_angles))
-
-            if not check_none_values[2]:
-                detunings = np.array(detunings, dtype=np.float).reshape((-1,))
-                input_array_lengths.append(len(detunings))
-
-            if not check_none_values[3]:
-                durations = np.array(durations, dtype=np.float).reshape((-1,))
-                input_array_lengths.append(len(durations))
-
-            # check all valid array lengths are equal
-            if max(input_array_lengths) != min(input_array_lengths):
-                raise ArgumentsValueError(
-                    "Rabi rates, Azimuthal angles, Detunings and Durations "
-                    "must be of same length",
-                    {
-                        "rabi_rates": rabi_rates,
-                        "azimuthal_angles": azimuthal_angles,
-                        "detunings": detunings,
-                        "durations": durations,
-                    },
-                )
 
-            valid_input_length = max(input_array_lengths)
-            if check_none_values[0]:
-                rabi_rates = np.zeros((valid_input_length,))
-            if check_none_values[1]:
-                azimuthal_angles = np.zeros((valid_input_length,))
-            if check_none_values[2]:
-                detunings = np.zeros((valid_input_length,))
-            if check_none_values[3]:
-                durations = np.ones((valid_input_length,))
-
-        self.rabi_rates = rabi_rates
-        self.azimuthal_angles = azimuthal_angles
-        self.detunings = detunings
-        self.durations = durations
+        # check if all non-None inputs have the same length
+        input_lengths = {
+            len(v)
+            for v in [rabi_rates, azimuthal_angles, detunings, durations]
```

Comment:
We also used to handle the case where these were multi-dimensional (we'd check the lengths of the flattened inputs). I guess that could be relevant if the user passes `[[1,2,3]]` or something... It's a little inefficient, but the simplest way I can think of to handle that case is:
 - change the `len(v)` to `len(np.array(v).size)`
 - call `.flatten()` when creating the arrays on 116--119

Reply:
SG.

> We also used to handle the case where these were multi-dimensional (we'd check the lengths of the flattened inputs). I guess that could be relevant if the user passes `[[1,2,3]]` or something... It's a little inefficient, but the simplest way I can think of to handle that case is:
> 
>     * change the `len(v)` to `len(np.array(v).size)`
IIUC, `np.array(v).size` would be fine for handling size?
>     * call `.flatten()` when creating the arrays on 116--119
SG

Reply:
> SG.
> 
> > We also used to handle the case where these were multi-dimensional (we'd check the lengths of the flattened inputs). I guess that could be relevant if the user passes `[[1,2,3]]` or something... It's a little inefficient, but the simplest way I can think of to handle that case is:
> > ```
> > * change the `len(v)` to `len(np.array(v).size)`
> > ```
> 
> IIUC, `np.array(v).size` would be fine for handling size?

D'oh.

Reply:
Done. The only issue I have now is that the docstring says all inputs are 1-D. Any reason why we want to support the multi-dimensional case here?

Reply:
It's probably not really necessary, I think it's generally just good if wherever possible we can avoid changing functionality (in a backwards incompatible way). In this case we can manage that without too much trouble, so I think it's worth it.